src/Helpers/QueriesRelationships.php

```diff
@@ -128,6 +130,26 @@ public function addHybridHas(Relation $relation, $operator = '>=', $count = 1, $
         return $this->whereIn($this->getRelatedConstraintKey($relation), $relatedIds, $boolean, $not);
     }
 
+    /**
+     * @param Relation $relation
+     *
+     * @return void
+     *
+     * @throws Exception
+     */
+    public function assertHybridRelationSupported(Relation $relation): void
+    {
+        if (
+            $relation instanceof HasOneOrMany
+            || $relation instanceof BelongsTo
+            || ($relation instanceof BelongsToMany && ! $this->isAcrossConnections($relation))
+        ) {
+            return;
+        }
+
+        throw new Exception(class_basename($relation) . ' is not supported for hybrid query constraints.');
```

Comment:
You can be more specific, this is a logic exception (developer fault).
```suggestion
        throw new \LogicException(class_basename($relation) . ' is not supported for hybrid query constraints.');
```